Homework

Get AVOCADO Price data from :

https://www.kaggle.com/neuromusic/avocado-prices

In [25]:
# import libraries
import pandas as pd
import numpy as np

In [26]:
# import data
data = pd.read_csv('Input/avocado.csv')

In [27]:
# exporatory data
data.head(5)

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

In [28]:
# Drop Unnamed: 0 column
data = data.drop(['Unnamed: 0'], axis = 1)

In [29]:
data.head()

Date  AveragePrice  Total Volume     4046       4225    4770  \
0  2015-12-27          1.33      64236.62  1036.74   54454.85   48.16   
1  2015-12-20          1.35      54876.98   674.28   44638.81   58.33   
2  2015-12-13          0.93     118220.22   794.70  109149.67  130.50   
3  2015-12-06          1.08      78992.15  1132.00   71976.41   72.58   
4  2015-11-29          1.28      51039.60   941.48   43838.39   75.78   

   Total Bags  Small Bags  Large Bags  XLarge Bags          type  year  region  
0     8696.87     8603.62       93.25          0.0  conventional  2015  Albany  
1     9505.56     9408.07       97.49          0.0  conventional  2015  Albany  
2     8145.35     8042.21      103.14          0.0  conventional  2015  Albany  
3     5811.16     5677.40      133.76          0.0  conventional  2015  Albany  
4     6183.95     5986.26      197.69          0.0  conventional  2015  Albany

**Attribute Description**

Date - The date of the observation

AveragePrice - the average price of a single avocado

type - conventional or organic

year - the year

Region - the city or region of the observation

Total Volume - Total number of avocados sold

4046 - Total number of avocados with PLU 4046 sold

4225 - Total number of avocados with PLU 4225 sold

4770 - Total number of avocados with PLU 4770 sold

**Summary Numeric Attribute**

In [30]:
data.describe()

AveragePrice  Total Volume          4046          4225          4770  \
count  18249.000000  1.824900e+04  1.824900e+04  1.824900e+04  1.824900e+04   
mean       1.405978  8.506440e+05  2.930084e+05  2.951546e+05  2.283974e+04   
std        0.402677  3.453545e+06  1.264989e+06  1.204120e+06  1.074641e+05   
min        0.440000  8.456000e+01  0.000000e+00  0.000000e+00  0.000000e+00   
25%        1.100000  1.083858e+04  8.540700e+02  3.008780e+03  0.000000e+00   
50%        1.370000  1.073768e+05  8.645300e+03  2.906102e+04  1.849900e+02   
75%        1.660000  4.329623e+05  1.110202e+05  1.502069e+05  6.243420e+03   
max        3.250000  6.250565e+07  2.274362e+07  2.047057e+07  2.546439e+06   

         Total Bags    Small Bags    Large Bags    XLarge Bags          year  
count  1.824900e+04  1.824900e+04  1.824900e+04   18249.000000  18249.000000  
mean   2.396392e+05  1.821947e+05  5.433809e+04    3106.426507   2016.147899  
std    9.862424e+05  7.461785e+05  2.439660e+05   17692.894652      0.939938  
min    0.000000e+00  0.000000e+00  0.000000e+00       0.000000   2015.000000  
25%    5.088640e+03  2.849420e+03  1.274700e+02       0.000000   2015.000000  
50%    3.974383e+04  2.636282e+04  2.647710e+03       0.000000   2016.000000  
75%    1.107834e+05  8.333767e+04  2.202925e+04     132.500000   2017.000000  
max    1.937313e+07  1.338459e+07  5.719097e+06  551693.650000   2018.000000

**Summary Categorical Attribute**

In [31]:
data.describe(include = ["O"])

Date          type  region
count        18249         18249   18249
unique         169             2      54
top     2015-12-27  conventional  Albany
freq           108          9126     338

1) Which region sold the largest amount of avocado?

In this region, where the biggest lot of sold avocado came form (4046, 4225, 4770)?

In [32]:
data.groupby(['region']).max(['Total Volume']).sort_values(['Total Volume'], ascending = False).head(1)

AveragePrice  Total Volume         4046         4225        4770  \
region                                                                      
TotalUS          2.09   62505646.52  22743616.17  20470572.61  2546439.11   

          Total Bags  Small Bags  Large Bags  XLarge Bags  year  
region                                                           
TotalUS  19373134.37  13384586.8  5719096.61    551693.65  2018

**Answer**

1. The region sold the largest amount is TotalUS sold 62,505,646.52.
2. The biggest came form lot 4046 are sold by 22,743,616.17.

2) Which region sold the smallest amount of avocado?

In this region, where the biggest lot of sold avocado came from (4046, 4225, 4770)?

In [33]:
data.groupby(['region']).min(['Total Volume']).sort_values(['Total Volume'], ascending = True).head(1)

AveragePrice  Total Volume  4046  4225  4770  Total Bags  \
region                                                                        
MiamiFtLauderdale          0.59         84.56  1.31   0.0   0.0       76.66   

                   Small Bags  Large Bags  XLarge Bags  year  
region                                                        
MiamiFtLauderdale       73.33         0.0          0.0  2015

**Answer**
1. The region sold the smallest amount is MiamiFtLauderdale sold by 84.56.
2. The biggest lot came from 4046 are sold 1.31.

3) Which region sold the highest price of avocado in average?

In [34]:
data.groupby(['region']).max(['AveragePrice']).sort_values(['AveragePrice'], ascending = False).head(1)

AveragePrice  Total Volume       4046        4225      4770  \
region                                                                      
SanFrancisco          3.25    1706251.05  560686.25  1113987.54  76966.69   

              Total Bags  Small Bags  Large Bags  XLarge Bags  year  
region                                                               
SanFrancisco   236417.93   231913.11    11823.61     10906.94  2018

**Answer**
1. The region are sold highest average price is SanFrancisco sold by 3.25.

4) Find the total amount of income ( Avg_Price * Total_Volume ) of each region.

In [37]:
income = data['AveragePrice'] * data['Total Volume']
data['income'] = income
data.head()

Date  AveragePrice  Total Volume     4046       4225    4770  \
0  2015-12-27          1.33      64236.62  1036.74   54454.85   48.16   
1  2015-12-20          1.35      54876.98   674.28   44638.81   58.33   
2  2015-12-13          0.93     118220.22   794.70  109149.67  130.50   
3  2015-12-06          1.08      78992.15  1132.00   71976.41   72.58   
4  2015-11-29          1.28      51039.60   941.48   43838.39   75.78   

   Total Bags  Small Bags  Large Bags  XLarge Bags          type  year  \
0     8696.87     8603.62       93.25          0.0  conventional  2015   
1     9505.56     9408.07       97.49          0.0  conventional  2015   
2     8145.35     8042.21      103.14          0.0  conventional  2015   
3     5811.16     5677.40      133.76          0.0  conventional  2015   
4     6183.95     5986.26      197.69          0.0  conventional  2015   

   region       income  
0  Albany   85434.7046  
1  Albany   74083.9230  
2  Albany  109944.8046  
3  Albany   85311.5220  
4  Albany   65330.6880

5) Let AVOCADO Average Weight: 4046 => 4 ounces, 4225 => 9 ounces, 4770 => 12 ounces

In [42]:
Average_Weight_4046 = data['4046'] / 4
Average_Weight_4225 = data['4225'] / 9
Average_Weight_4770 = data['4770'] / 12
data['Average_Weight_4046'] = Average_Weight_4046
data['Average_Weight_4225'] = Average_Weight_4225
data['Average_Weight_4770'] = Average_Weight_4770
data.head()

Date  AveragePrice  Total Volume     4046       4225    4770  \
0  2015-12-27          1.33      64236.62  1036.74   54454.85   48.16   
1  2015-12-20          1.35      54876.98   674.28   44638.81   58.33   
2  2015-12-13          0.93     118220.22   794.70  109149.67  130.50   
3  2015-12-06          1.08      78992.15  1132.00   71976.41   72.58   
4  2015-11-29          1.28      51039.60   941.48   43838.39   75.78   

   Total Bags  Small Bags  Large Bags  XLarge Bags          type  year  \
0     8696.87     8603.62       93.25          0.0  conventional  2015   
1     9505.56     9408.07       97.49          0.0  conventional  2015   
2     8145.35     8042.21      103.14          0.0  conventional  2015   
3     5811.16     5677.40      133.76          0.0  conventional  2015   
4     6183.95     5986.26      197.69          0.0  conventional  2015   

   region       income  Average_Weight_4046  Average_Weight_4225  \
0  Albany   85434.7046              259.185          6050.538889   
1  Albany   74083.9230              168.570          4959.867778   
2  Albany  109944.8046              198.675         12127.741111   
3  Albany   85311.5220              283.000          7997.378889   
4  Albany   65330.6880              235.370          4870.932222   

   Average_Weight_4770  
0             4.013333  
1             4.860833  
2            10.875000  
3             6.048333  
4             6.315000

6) Normally, the customers buy the avocados by unit or in a bags ?

In [60]:
Buy_Unit = data['4046'] + data['4225'] + data['4770']
data['Buy_Unit'] = Buy_Unit
data[['Buy_Unit', 'Total Bags']].sum()

Buy_Unit      1.115019e+10
Total Bags    4.373176e+09
dtype: float64

**Answer**
1. Customers buy avocados by unit 11,150,190,000 more than bags 4,373,176,000.

**Optional**

7) Which month is the best time to sell avocado

8) Base on the data in 2015, is it true that the region which higher population can sell more avocados?